# Applying integrative multi-omics to TAA

In [1]:
import pandas as pd
import numpy as np
from pathlib import Path

## VCF file
Opening and analysing the variant file containing variant sites exclusive to SBC10.

In [4]:
%%bash
VCF_PATH=../results/sift4g/SBC10.private.sift4g.vcf.gz
../docker/run.sh python3 ../workflow/scripts/genomics_scoring.py -i $VCF_PATH -o ../results/gene_score/ -g ../resources/annot/GCF_000003195.3_Sorghum_bicolor_NCBIv3_genomic.gff.gz

[genomics_scoring] Parsing VCF: ../results/sift4g/SBC10.private.sift4g.vcf.gz
[genomics_scoring]   727,988 (variant, gene) rows
[genomics_scoring] Written → ../results/gene_score/SBC10.variant.tsv  (727,988 variant annotations)
[genomics_scoring] Loading gene lengths: ../resources/annot/GCF_000003195.3_Sorghum_bicolor_NCBIv3_genomic.gff.gz
[genomics_scoring]   32,458 genes with length
[genomics_scoring] Written → ../results/gene_score/SBC10.gene_max.tsv  (36,912 genes)
[genomics_scoring] Written → ../results/gene_score/SBC10.gene_sum.tsv  (36,912 genes)
[genomics_scoring] Written → ../results/gene_score/SBC10.gene_max_norm.tsv  (36,912 genes)
[genomics_scoring] Written → ../results/gene_score/SBC10.gene_sum_norm.tsv  (36,912 genes)
[genomics_scoring] Written → ../results/gene_score/SBC10.scores_summary.png
[genomics_scoring] Written → ../results/gene_score/SBC10.variant_scores.png


Extra genes (36,912 genes), more than the genes loaded in the GFF file (32,458), accounts for intergenic region (GeneA-GeneB), that is a combination between 2 adjacent genes. The script detects them as genes, when in fact these are not, but quite a pseudo-feature.

### Choosing the most appropriate merging method out of the four

In [7]:
df_scored = pd.read_csv("../results/gene_score/SBC10.variant.tsv", sep="\t")


,chrom,pos,ref,alt,qual,filter,gt,ann_allele,ann_effect,ann_impact,...,sift_variant_type,sift_aa_change,sift_aa_pos,sift_score,sift_median,sift_num_seqs,sift_allele_type,sift_prediction,sift_score_c,score
0,NC_012870.2,1619,G,A,54.540001,NaN,1|0,A,upstream_gene_variant,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
1,NC_012870.2,1619,G,A,54.540001,NaN,1|0,A,intergenic_region,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
2,NC_012870.2,5784,A,"AGG,AG",25.700001,NaN,1/2,AG,downstream_gene_variant,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
3,NC_012870.2,5784,A,"AGG,AG",25.700001,NaN,1/2,AG,intergenic_region,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
4,NC_012870.2,6934,A,G,31.510000,NaN,1|0,G,downstream_gene_variant,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
727983,NC_012879.2,61233619,T,G,27.490000,NaN,1/1,G,intergenic_region,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
727984,NC_012879.2,61233644,A,AG,35.650002,NaN,1/1,AG,upstream_gene_variant,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
727985,NC_012879.2,61233644,A,AG,35.650002,NaN,1/1,AG,intergenic_region,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
727986,NC_012879.2,61233676,G,GT,32.020000,NaN,1/1,GT,upstream_gene_variant,MODIFIER,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.125
